xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCC.cpp

```diff
@@ -11,9 +11,95 @@
 #include "DVDDemuxUtils.h"
 #include "cores/VideoPlayer/DVDCodecs/Overlay/contrib/cc_decoder708.h"
 #include "cores/VideoPlayer/Interface/TimingConstants.h"
+#include "utils/ColorUtils.h"
+#include "utils/StringUtils.h"
 
 #include <algorithm>
 
+namespace
+{
+/*!
+ * \brief Color formats for CC color conversion
+ */
+enum class ColorFormat
+{
+  RGB,
+  ARGB
+};
+
+/*!
+ * \brief Given the color code from the close caption decoder, returns the corresponding
+ * color in rgb format (striping the alpha channel from argb)
+ * 
+ * \param[in] ccColor - a given color from the cc decoder
+ * \param[in] format - the color format
+ * \return the corresponding Color in rgb
+ */
+constexpr UTILS::COLOR::Color CCColorConversion(const uint8_t ccColor, ColorFormat format)
+{
+  UTILS::COLOR::Color color = UTILS::COLOR::NONE;
+  switch (ccColor)
+  {
+    case WHITE:
+      color = UTILS::COLOR::WHITE;
+      break;
+    case GREEN:
+      color = UTILS::COLOR::GREEN;
+      break;
+    case BLUE:
+      color = UTILS::COLOR::BLUE;
+      break;
+    case CYAN:
+      color = UTILS::COLOR::CYAN;
+      break;
+    case RED:
+      color = UTILS::COLOR::RED;
+      break;
+    case YELLOW:
+      color = UTILS::COLOR::YELLOW;
+      break;
+    case MAGENTA:
+      color = UTILS::COLOR::MAGENTA;
+      break;
+    case BLACK:
+      color = UTILS::COLOR::BLACK;
+      break;
+    default:
+      break;
+  }
+
+  if (format == ColorFormat::RGB)
+    return color & ~0xFF000000;
+
+  return color;
+}
+
+/*!
+ * \brief Given the current buffer cc text and cc style attributes, apply the modifiers
+ * \param[in,out] ccText - the text to display in the caption
+ * \param[in] ccAttributes - The attributes (italic, color, underline) for the text
+ */
+void ApplyStyleModifiers(std::string& ccText, const cc_attribute_t& ccAttributes)
+{
+  // Apply style modifiers to CC text
+  if (ccAttributes.italic > 0)
+  {
+    ccText = StringUtils::Format("<i>{}</i>", ccText);
+  }
+  if (ccAttributes.underline > 0)
+  {
+    ccText = StringUtils::Format("<u>{}</u>", ccText);
+  }
+  if (ccAttributes.foreground != WHITE)
+  {
+    ccText = StringUtils::Format(
+        "<font color=#{}>{}</u>",
+        StringUtils::Format("{:06X}", CCColorConversion(ccAttributes.foreground, ColorFormat::RGB)),
+        ccText);
+  }
+}
+} // namespace
+
```

Comment:
since it is only for 608 and not usable for 708 why not move all this block in to ccdecoder itself or in a separate helper header?

Reply:
because it's common to the two of them (it's just not implemented for 708).
Plus the decoder itself has fixed size for rows and text (CC_ROWS*CC_COLUMNS) so you don't want to change any data there. The modifiers are pretty much something that belong to the context of the component actually feeding overlays (not the decoder)

Reply:
Also they share the same colors. From CTA708:
- At minimum decoders shall implement the following colors: white, black, red, green, blue, yellow, magenta and cyan.

There's additional colors (up to 22) in the standard (table 31).